lib/vrl/stdlib/src/encode_json.rs

```diff
@@ -70,37 +70,37 @@ mod tests {
         bytes {
             args: func_args![value: r#"hello"#],
             want: Ok(r#""hello""#),
-            tdef: TypeDef::new().infallible().bytes(),
```

Comment:
I've removed the `new` impl because it was a potential for misusing the type-def API. That is, it would instantiate a type def without any `kind` information, which isn't a valid state to be in.